storage/innobase/row/row0mysql.cc

```diff
@@ -2135,6 +2210,14 @@ row_update_for_mysql(row_prebuilt_t* prebuilt)
 		prebuilt->table->stat_modified_counter++;
 	}
 
+	if (node->table->versioned() &&
+	    (node->versioned || node->vers_delete ||
+	     // TODO: improve this check (check if we touch only
+	     // unversioned fields in foreigh table)
+	     node->foreign)) {
+		trx->vers_update_trt = true;
+	}
+
```

Comment:
I think that the proper fix would be to do the update as follows:

1. In the old row, set end_time = trx_id
2. Insert a new row for the updated data, with start_time = trx_id and end_time=MAX.

Hopefully also FOREIGN KEY operations would be simpler and faster in this way. OK, if cascaded operation is requested, then some additional logic would be needed to implement that.

Reply:
Scheduled in https://github.com/tempesta-tech/mariadb/issues/338 on next release. If you feel that it's important for this merge, feel free to object.